### ISSUE TITLE
How can I test the model with my custom bin-picking scene?

### ISSUE THREAD

**Issue by aitorgutierrez** [BerkeleyAutomation/sd-maskrcnn#41]
Hi!

I will like to validate the model with my bin-picking scene (my custom objects). I am using a Real Sense D435, and I am able to get the RGB image (color_ims) and the depth images (modal_segmasks).

How can generate the depth_ims from  depth images? How can I generate de train_indices.npy?

I would like to test, how it is working the model on my scene, but I do not know how to run the inference.

Best regards

**Comment by mjd3** [comment]
@aitorgutierrez 
You can definitely use your own images with your own objects. RGB and depth images can be taken directly with your camera (color_ims and depth_ims). If you're using the pretrained model, then you'll want to size these accordingly (384x512 padded to 512x512). To generate the labels (modal_segmasks), you can use a tool like [this](https://github.com/yuyu2172/image-labelling-tool). For benchmarking, your indices.npy file will just be an array of all image indices. Then, you should be able to just run the benchmark.py script in the tools directory. Let me know if you run into any issues!

**Comment by aitorgutierrez** [comment]
Thank you for the fast reply. I was able to borrow a Photoneo. Did you get your images using the Phoxi control software? It is not possible to get a RGB image, but I think that I can use 2D single channel image.  How can generate depth_ims with this software? Is it to save the deph image as screenshot option? Why I need to label the image using the image-labelling-tool?I only want to test how the model is segmenting my image?What should be the input?

Best regards

**Comment by mjd3** [comment]
Sorry I misunderstood what you want to do. If you just want to use the model and not benchmark on your own labeled images, then you can use the [ROS wrapper](https://github.com/photoneo/phoxi_camera) to read depth images from the camera and directly feed these images into the pretrained model to get segmentations. The ```tools/benchmark.py``` script should give you a good idea of how to use the model. Let me know if you have any questions!

**Comment by ltorquato4** [comment]
Hi @aitorgutierrez ,
I am using Intel RealSense D435 as well and I wanted to implement SD-RCNN in my pipeline. 
@aitorgutierrez if you want to test with the D435 also you can use this [ ROS wrapper](https://github.com/IntelRealSense/realsense-ros/releases).

**Comment by ltorquato4** [comment]
Dear @mjd3, 
I am doing my bachelor thesis based on GQ-CNN and I want to use SD-RCNN in order to improve the results (Q-value). 
I read the SD-RCNN's paper and it suggests that the model should generalize well to other camera intrinsics and poses and to novel objects as well. 
However, considering that I want to have reliable results, would you recommend that I train the model again, with my camera (Intel RealSense D435) and including my objects in the training set (+ WISDOM-Sim)?
Thank you for you attention.

**Comment by mjd3** [comment]
Hi @ltorquato4, glad to hear you are using the project. For a different camera setup, I would first test with the setup that you have. If you find that your camera intrinsics/extrinsics are significantly different than those used for the pretrained model (see [here](https://github.com/BerkeleyAutomation/sd-maskrcnn/blob/a53826f6ee44960f98cf42d827aa50d1b5731387/cfg/partials/states.yaml#L62) for the parameters used in training), then I would suggest training a model from scratch with camera parameters that more closely match your setup. You can check out the dataset generation part of the Readme for more info on that and I am happy to answer any questions as well!

**Comment by ltorquato4** [comment]
Hi @mjd3 , in the past days I tried to test the benchmark tool in the wisdom dataset, but unfortunately I run into some errors.
I am using a virtualenv just for SDMaskRCNN with python 3.6 and I installed all packages from the requirements. Following the tensorflow requirements the only one that didn't crash (because of other versions of libraries required) was 1.12.3 (tensorboard, tensorflow, tensorflow-gpu) and Keras 2.3.1, Keras-Applications 1.0.8 and Keras-Preprocessing 1.1.0. 
Running the benchmark tool the first error that appears in the line 41 from the Code is "module 'tensorflow.python.keras.backend' has no Attribute 'get_graph'". I search the solution for it and it seems that instead of using `from keras.backend.tensorflow_backend import set_session`  it should be `from tensorflow.keras.backend import set_session` . I Changed it, but I got more similar errors in the following lines and also from the MaskRCNN codes. So I thought that it is not possible that I have to change all them. 
Do you have any idea of which could be the problem? Or should I just keep going and changing the codes?

**Comment by mjd3** [comment]
Hey @ltorquato4, sorry for the late response. When I built this repo, I was using python 3.5 with tensorflow 1.10 and keras 2.2, for reference. I haven't tested with other versions of python/tf/keras, but if you figure out a good way to make the code compatible with these versions, I'd happily accept a PR!! I'm sure there will be many incompatibilities like this unfortunately, as these libraries do change fairly quickly.

**Comment by mjd3** [comment]
I've also been working on creating a Docker image for this project, but it's a little bit on the back burner for now; probably will be done in March/April if that's a better solution for you!
